Add `push` and `pull` steps for Azure Blob Storage

desertaxle opened the pull request:
<!-- Thanks for contributing 🎉! Please ensure the title neatly summarizes the proposed changes. -->

<!-- Overview -->
- Adds `push_to_azure_blob_storage` and `pull_from_azure_blob_storage` project steps to allow Azure Blob Storage to be used for deployment storage and retrieval. New steps follow the pattern established for existing `push` and `pull` steps.
- Cleans up docstrings and field description for `AzureBlobStorageCredentials`
- Adds logo and documentation URLs to `AzureContainerWorker`

Closes #82 
Closes #83

### Example
<!-- A code blurb is best. Changes to features should include an example that is executable by a new user. -->
```yaml
prefect_version: ...
name: ...

push:
    - prefect_azure.deployments.steps.push_to_azure_blob_storage:
        requires: prefect-azure
        container: my-container
        folder: my-project
        credentials: "{{ prefect.blocks.azure-blob-storage-credentials.dev-credentials }}"

pull:
    - prefect_azure.deployments.steps.pull_from_azure_blob_storage:
        requires: prefect-azure
        container: "{{ container }}"
        folder: "{{ folder }}"
        credentials: "{{ prefect.blocks.azure-blob-storage-credentials.dev-credentials }}"
```

### Screenshots
<!--
Any relevant screenshots
  - The updated docs page from `mkdocs serve`.
  - Output from running the example.
  - Service integration test results.
-->
![localhost_8000_prefect-azure_deployments_steps_](https://github.com/PrefectHQ/prefect-azure/assets/12350579/1f45be93-67cb-4a22-b8fa-a3a02fdf2bc9)

### Checklist
<!-- These boxes may be checked after opening the pull request. -->

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-azure/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [x] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-azure/blob/main/CHANGELOG.md)

serinamarie commented on tests/projects/test_steps.py:
```diff
@@ -0,0 +1,348 @@
+        self, tmp_files: Path, container_client_mock: MagicMock
+    ):
+        container = "test-container"
+        folder = "test-folder"
```
are trailing slashes handled?

serinamarie reviewed (commented):
Nice! 

For the tests I was wondering regarding folders:
1. Are trailing slashes in the `folder` arg handled?
2. Should we test that subfolders are properly read from/to?
3. It seems like all of these tests include a `folder` arg. Should we test not including a folder?

desertaxle commented:
@serinamarie I added some test coverage based on your suggestions and removed references to 'projects' in this PR, so it should be ready for another review.

jakekaplan commented on prefect_azure/deployments/steps.py:
````diff
@@ -0,0 +1,196 @@
+            credentials: "{{ prefect.blocks.azure-blob-storage-credentials.dev-credentials }}"
+    ```
+
+For more information about using deployment steps, check out out the Prefect [docs](https://docs.prefect.io/latest/concepts/projects/#the-prefect-yaml-file).
````
Is this the right link?

jakekaplan commented on prefect_azure/deployments/steps.py:
```diff
@@ -0,0 +1,196 @@
+        - prefect_azure.deployments.steps.push_to_azure_blob_storage:
+            requires: prefect-azure[blob_storage]
+            container: my-container
+            folder: my-project
```
not sure if you want to call it something different than `my-project` to differentiate from "projects" at all 🤷‍♂️
